src/StringUtils.cpp

```diff
@@ -14,6 +14,259 @@
 #include <numeric>
 #include <sstream>
 
+template<>
+char NAS2D::detail::rangeCheckHelper(const std::string& value)
+{
+	if (value.empty())
+	{
+		throw std::invalid_argument("rangeCheckHelper string must be non-empty.");
+	}
+	return value[0];
```

Comment:
I'm still on the fence about using quoted string values for this, rather than small range integers.

If we're going to use quoted string values, this check should at least be changed so it checks for exactly one letter strings, rather than not 0 length strings.

Reply:
Non-signed and non-unsigned `char`s are understood to be single character values. If you want small range integers, use `signed char` or `unsigned char`.

Reply:
I understand where you're coming from. Though I would like to point out that `'a'` and `"a"` are different types in C++. To specify a single `char`, you use the `'a'` syntax. If you use `"a"` you get a `char[]` (which decays to `char*`). Syntactically, the `char` type is compatible with single character literals of the form `'a'` and with integer literals of the form `97`.

Additionally it's fairly common practice to use `char` to hold byte arrays for binary data. It's maybe not the best practice, but it is quite common.

----

Regardless of syntax used for literals, a `char` type can only hold 1 character, so the check should be for a string of `size` 1.